### ISSUE TITLE
Execute tests against ERB html5 backend

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#839]
Since Asciidoctor.js relies on the ERB html5 backend to be up to date, we should run the test suite against these templates as well as the built-in ones.

As a bonus, we should run the test against the Haml and Slim html5 backends as well. Haml should be straightforward. Slim will be tougher since it drops a lot of whitespace in the output.

**Comment by mojavelinux** [comment]
This is now handled by doctest in a much more robust way, thanks to @jirutka. The tests run as part of the asciidoctor-backends repository.

**Comment by jirutka** [comment]
Eh, we’re not testing ERB templates, just Slim and Haml. ERB templates are outdated, I fixed and updated only Slim and Haml templates.

I consider ERB templates deprecated, ERB is IMO totally inappropriate for Asciidoctor backend, these templates are a huge unmaintainable mess.

Is there any problem with using Slim in Asciidoctor.js?

**Comment by mojavelinux** [comment]
> we’re not testing ERB templates, just Slim and Haml.

I wasn't clear in my comment that I was thinking about Slim and Haml. The overall intent of the issue is that asciidoctor-backends is being tested in some way. That's now true. What we chose to test is up to asciidoctor-backends and doctest.

>  ERB is IMO totally inappropriate for Asciidoctor backends, these templates are a huge unmaintainable mess.

I don't disagree with you. They exist to demonstrate it's possible to use ERB (and because in the beginning we didn't have haml and slim templates). I wonder if we should consider trimming ERB back to a few samples (such as paragraph) to show it's possible and just leave the rest out.

> Is there any problem with using Slim in Asciidoctor.js?

I don't think Tilt compiles with Opal yet. I think it would be cleaner if we found a way to be able to use Jade from Asciidoctor.js. The Tilt layer is pretty abstracted away, so we might be able to get it working without too much trouble.
